primitives/src/vrf.rs

```diff
@@ -232,3 +250,155 @@ pub mod sortition {
 		}
 	}
 }
+
+const ENGINE_ID: &[u8; 4] = b"COTS";
+
+pub(super) fn transcript_data(pre_hash: H256, epoch: u64, task_id: H256) -> VRFTranscriptData {
+	VRFTranscriptData {
+		label: ENGINE_ID,
+		items: vec![
+			("epoch", VRFTranscriptValue::U64(epoch)),
+			("task id", VRFTranscriptValue::Bytes(task_id.encode())),
+			("pre hash", VRFTranscriptValue::Bytes(pre_hash.encode())),
```

Comment:
I understand why the epoch and task id are part of the transcript, but I feel like I'm missing the intuition behind the pre hash. Why do we want to include it in the transcript?

I may also just be confused because I'm not sure what the `pre_hash` is lol

Reply:
Same reasoning behind including the pre_hash in a PoW seal; without that source of randomness, an attacker would know in advance who is going to be sampled and makes the sampled nodes easy attack vectors. Or in theory, you could forge a public key guaranteed to be sampled for a given transcript.